apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts

```diff
@@ -42,25 +43,29 @@ export class InAppOutputRendererUsecase {
   }
 
   private buildActionIfAllPartsAvailable(action?: InAppActionType) {
-    if (!(action && action.label && action.redirect && action.redirect.url && isValidURL(action.redirect.url))) {
+    if (!(action && action.label)) {
       return undefined;
     }
+    if (!action.redirect) {
+      return {
+        label: action.label,
+      };
+    }
+    if (!action.redirect.url) {
+      return {
+        label: action.label,
+      };
+    }
+    if (isValidUrlForActionButton(action.redirect.url)) {
```

Comment:
same here should we validate `redirect.target`?

Reply:
i'll add it